src/lexer/lexer.sml

```diff
@@ -15,6 +15,7 @@ struct
         in
             ErrorMsg.reset();
             Newline.reset();
+            SrcComment.reset();
```

Comment:
I think we should reset everything here. All this state is really not great, so the least we can do is try to manage it by being consistent about where it gets reset.